lightning-invoice/src/de.rs

```diff
@@ -600,7 +601,7 @@ impl FromBase32 for PrivateRoute {
 			channel_id.copy_from_slice(&hop_bytes[33..41]);
 
 			let hop = RouteHintHop {
-				src_node_id: PublicKey::from_slice(&hop_bytes[0..33])?,
+				src_node_id: hop_bytes[0..33].try_into().map_err(|_| ParseError::InvalidSliceLength("PrivateRoute::from_base32()".into()))?,
```

Comment:
I think I'd rather we keep lightning-invoice as a `PublicKey` and then just `serialize()` to get the `[u8; 33]` in `get_route` - we shouldn't have to worry too much about invoice parsing performance there, I think. Then we can also avoid touching `RouteHintHop` which should cut down the diff size a good chunk.

Reply:
Yeah, that would simplify things. Thanks, will update! Should make it much easier to review then too.